### PR TITLE
176: Modify PCA backend to further match the requirements

### LINKED ISSUE #176: Modify PCA backend to further match the requirements
In my previous implementation, I did not have the functionality to:
1. get variance explained by principal components
2. select a specific principal from a PCA-transformed dataset.

I am working on the fix to further match the requirement

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #176 

# What was the issue
There was no function to:
1.  select a principal component from a PC-transformed data set
2. calculate the variance explained by principal components

# What was done and why
Add two functions:
- calculateVarianceExplained()
- selectPrincipalComponent()

Remove ClassicPCA and its tests to better maintain the PCA backend

# How it was tested
- Test against the unit tests

### CODE REVIEW

File: my-webxr-app/src/utils/PcaClassic.tsx
Comment (Subzeero) [inline my-webxr-app/src/utils/PcaClassic.tsx:1]: So classic PCA was just another method to do PCA? The covariance PCA is a better version?

File: my-webxr-app/src/utils/PcaCovariance.tsx [hunk 1]
```diff
@@ -70,29 +69,67 @@ export function computeEigenvectorsFromCovarianceMatrix(covarianceMatrix: Matrix
...
+ */
+export function calculateVarianceExplained(covarianceMatrix: Matrix): number[] {
+  try {
+    const eigenvalues = computeEigenvaluesFromCovarianceMatrix(covarianceMatrix);
+    const totalVariance = eigenvalues.reduce((a, b) => a + b, 0);
+    return eigenvalues.map((value) => value / totalVariance);
+  } catch (e) {
+    // may need to surface this error somewhere
+    // console.error(`An error occurred during calculating variance explained: ${e}`);
+    return [];
```
Comment (Subzeero) [inline my-webxr-app/src/utils/PcaCovariance.tsx:86]: Is this a fatal error? As in one that means conducting PCA failed? If so, we will probably have to inform the user if such a case occurs. Would throwing an Exception be reasonable? + logging once we get that implemented.

**Review by Subzeero — COMMENTED** [review]
Good documentation! I've just got a few questions

File: my-webxr-app/src/utils/PcaClassic.tsx
Comment (LongQuanNguyen) [reply my-webxr-app/src/utils/PcaClassic.tsx:1]: Yep, just another method of doing PCA. But I am more familiar with Covariant method.

File: my-webxr-app/src/utils/PcaCovariance.tsx (hunk 1 above)
Comment (LongQuanNguyen) [reply my-webxr-app/src/utils/PcaCovariance.tsx:86]: Yes, something like a null data point get past the filtering before the PCA.

Comment (LongQuanNguyen) [reply my-webxr-app/src/utils/PcaCovariance.tsx:86]: We could catch the errors and rethrown them here. As right now I just don't want this to break on the user end and I am not sure how we going to notify user yet.

**Review by matthew-buglass — APPROVED** [review]

**Pull request merged**